src/unix/sock_ctrl_msg.rs

```diff
@@ -113,8 +125,12 @@ fn raw_sendmsg<D: IntoIovec>(fd: RawFd, out_data: &[D], out_fds: &[RawFd]) -> Re
 
     if !out_fds.is_empty() {
         let cmsg = cmsghdr {
-            // SAFETY: We ensure that the input value does not exceed the range of c_uint. c_uint to usize is a safe conversion
-            cmsg_len: unsafe { CMSG_LEN(size_of_val(out_fds).try_into().unwrap()) as usize },
+            #[cfg(any(target_os = "linux", target_os = "android"))]
+            // SAFETY: We ensure that the input value does not exceed the range of c_uint.
+            cmsg_len: unsafe { CMSG_LEN(size_of_val(out_fds).try_into().expect("size_of_value(out_fds) exceeds u32 range")) as usize },
+            #[cfg(target_os = "macos")]
+            // SAFETY: We ensure that the input value does not exceed the range of c_uint.
+            cmsg_len: unsafe { CMSG_LEN(size_of_val(out_fds).try_into().expect("size_of_value(out_fds) exceeds u32 range")) },
```

Comment:
Maybe we can call CMSG_LEN.. once, assign that to a variable and just do the cast like:
```rust
let cmsg_len = unsafe { CMSG_LEN(size_of_val(out_fds).try_into().expect("size_of_value(out_fds) exceeds u32 range"))

...
cmsg_len:  if cfg!(target_os = "macos") { cmsg_len } else { cmsg_len as usize}
```

or something like this

Reply:
The statement if should return the same type. On macos, cmsg_len is u32, but cmsg_len as usize will return usize. So rust compiler will report an error here